src/coreclr/jit/gentree.cpp

```diff
@@ -17757,7 +17757,15 @@ CORINFO_CLASS_HANDLE Compiler::gtGetClassHandle(GenTree* tree, bool* pIsExact, b
         case GT_CALL:
         {
             GenTreeCall* call = tree->AsCall();
-            if (call->IsInlineCandidate())
+            if (call->gtFlags & CORINFO_FLG_JIT_INTRINSIC)
+            {
+                if (lookupNamedIntrinsic(call->gtCallMethHnd) == NI_System_Array_Clone)
+                {
+                    objClass = gtGetClassHandle(call->gtCallThisArg->GetNode(), pIsExact, pIsNonNull);
+                    break;
+                }
+            }
+            else if (call->IsInlineCandidate())
```

Comment:
```suggestion
            if (call->IsInlineCandidate())
```
Probably unlikely, but we could have a jit intrinsic that's also an inline candidate.